source/slang/slang-parser.cpp

```diff
@@ -4972,6 +4972,13 @@ static void CompleteDecl(
                     staticConstDecl->nameAndLoc = enumCase->nameAndLoc;
                     addModifier(staticConstDecl, parser->astBuilder->create<HLSLStaticModifier>());
                     addModifier(staticConstDecl, parser->astBuilder->create<ConstModifier>());
+
+                    // Copy visibility modifiers from the enum declaration
+                    if (auto visibilityModifier = modifiers.findModifier<VisibilityModifier>())
+                    {
+                        addModifier(staticConstDecl, visibilityModifier);
```

Comment:
I think you want to create a new visibility modifier of the same type and add that to `staticConstDecl`, instead of adding the existing modifier instance found on `modifiers`, because Modifier itself form a linked list. We don't want to share anything after the visibility modifier between the static const and the enum decl.